src/components/map/layers/RoutePathLinkLayer.tsx

```diff
@@ -47,29 +47,32 @@ class RoutePathLayer extends Component<RoutePathLinkLayerProps> {
     }
 
     private renderNodes() {
+        const openPopup = (node: INode) => () => {
+            this.props.popupStore!.showPopup(node);
+        };
         const routePathLinks = this.props.routePathLinks;
-        return routePathLinks.map((routePathLink, index) => {
-            return (
-                <div key={index}>
-                    <NodeLayer
-                        key={`${routePathLink.startNode.id}`}
+        const nodes = routePathLinks
+            .map((routePathLink, index) => {
+                return (
+                    <NodeMarker
+                        key={`${routePathLink.startNode.id}-${index}`}
                         node={routePathLink.startNode}
                         isDisabled={routePathLink.startNodeType === NodeType.DISABLED}
                         isTimeAlignmentStop={routePathLink.isStartNodeTimeAlignmentStop}
+                        onContextMenu={openPopup(routePathLink.startNode)}
                     />
-                    { index === routePathLinks.length - 1 &&
-                        <NodeLayer
-                            key={`${routePathLink.endNode.id}`}
-                            node={routePathLink.endNode}
-                            /* hardcoded because last node doens't have this data */
-                            isDisabled={false}
-                            /* hardcoded because last node doens't have this data */
-                            isTimeAlignmentStop={false}
-                        />
-                    }
-                </div>
-            );
-        });
+                );
+            });
+        nodes.push(
+            <NodeMarker
+                key={`${routePathLinks[routePathLinks.length - 1]
```

Comment:
Now we specify routePathLinks[routePathLinks.length - 1] three times in this block. Maybe we could create a variable named lastRoutePathLink, and use that to not duplicate fetching of the last link?